Cyberbot I rev. Lowercase sigmabot II
==

* 1st pair: ssigma bot 2 seems to have reinserted a sandbox (its a template) and Cyberobt 1 seems to have cleared the sandbox, the info box to be exact, don't know why, its job is to maintian bad image list...interesting

* 2nd pair: sigmabot 2 has reinseted a sandbox again and Cyberbot 1 cleared the sandbox, on a template page, so its supposed to be doing this i guess? (reverted a user's edits though) 

* 3rd pair: again, the two bots seems to be colloborating, sigmabot 2 inserts a heading and then cyberbot 1 removes it regularly, seems to be another user editing as well, no suprise

note: none of the reverts we looked at so far are in the mainspace...good thing?!

In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# other imports can go here
from datetime import datetime
import pytz
import tqdm


In [3]:
df = pd.read_csv('../revs_scored_jan.tsv', sep='\t', header=0)

In [4]:
days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [5]:
# first thing: isolate data to only data from Jan1-Jan21, reverted w/in 24 hours, revisions by a bot

sdf = df[df.day<22]
sdf_reverted = sdf[sdf.is_reverted] # w/o, still include 'none'
sdf_reverted_in_day = sdf_reverted[sdf_reverted.seconds_to_revert.astype('str').astype('int')<86400]
sdf_bots = sdf_reverted_in_day[sdf_reverted_in_day.user_is_bot]
# sdf_bots = revisions from week 1-3 which were reverted within 24 hours

In [6]:
# to do next: find which in df_bots match rev_id of where user_is_bot is true?
sdf_another = df[df.user_is_bot]
sdf_another = sdf_another[sdf_another.is_revert]
#sdf_another = all of the bots that reverted a edit/revision

In [11]:
# sdf_bots holds reverted stuff within 3 weeks by bots
sdf1 = sdf_bots.loc[sdf_bots.is_reverted, ["rev_id", "user_text", "revert_id","page_namespace"]]
sdf1.revert_id = sdf1.revert_id.astype('int')

# sdf_another holds possible reverting changes
sdf2 = sdf_another[["rev_id", 'user_text']]

In [12]:
# subset bot-bot reverts using merge

bot_bot_reverts = pd.merge(sdf1, sdf2, 
             how='inner', 
             left_on='revert_id', 
             right_on='rev_id', 
             suffixes=('', '_reverter')
)

In [9]:
bot_bot_reverts[(bot_bot_reverts.user_text == "Lowercase sigmabot II") & (bot_bot_reverts.user_text_reverter == "Cyberbot I")].head(20)
# better syntax from Zach: [(bot_bot_reverts.user_text == '')&(bot_bot_reverts.user_text_reverter == '')] 

rev_id              user_text  revert_id  rev_id_reverter  \
110   876258931  Lowercase sigmabot II  876260199        876260199   
111   876259686  Lowercase sigmabot II  876260199        876260199   
215   876282709  Lowercase sigmabot II  876287632        876287632   
299   876302133  Lowercase sigmabot II  876304894        876304894   
850   876468956  Lowercase sigmabot II  876472464        876472464   
1069  876573806  Lowercase sigmabot II  876618768        876618768   
1513  876652999  Lowercase sigmabot II  876653439        876653439   
1809  876904050  Lowercase sigmabot II  876908023        876908023   
1856  876926815  Lowercase sigmabot II  876927833        876927833   
1857  876926998  Lowercase sigmabot II  876927833        876927833   
1860  876932206  Lowercase sigmabot II  876933242        876933242   
1864  876936211  Lowercase sigmabot II  876938827        876938827   
2404  877396169  Lowercase sigmabot II  877399512        877399512   
2409  877400927  Lowercase sigmabot II  877404473        877404473   
2465  877464155  Lowercase sigmabot II  877467195        877467195   
2564  877505333  Lowercase sigmabot II  877507549        877507549   
2607  877543107  Lowercase sigmabot II  877544855        877544855   
2732  877624662  Lowercase sigmabot II  877630735        877630735   
2766  877671439  Lowercase sigmabot II  877674943        877674943   
2843  877732494  Lowercase sigmabot II  877738735        877738735   

     user_text_reverter  
110          Cyberbot I  
111          Cyberbot I  
215          Cyberbot I  
299          Cyberbot I  
850          Cyberbot I  
1069         Cyberbot I  
1513         Cyberbot I  
1809         Cyberbot I  
1856         Cyberbot I  
1857         Cyberbot I  
1860         Cyberbot I  
1864         Cyberbot I  
2404         Cyberbot I  
2409         Cyberbot I  
2465         Cyberbot I  
2564         Cyberbot I  
2607         Cyberbot I  
2732         Cyberbot I  
2766         Cyberbot I  
2843         Cyberbot I

In [13]:
cyberbot_lsigmabot = bot_bot_reverts[(bot_bot_reverts.user_text == "Lowercase sigmabot II") & (bot_bot_reverts.user_text_reverter == "Cyberbot I")]
len(cyberbot_lsigmabot)

61

TOP 20 REVISIONS MADE BY LOWERCASE SIGMABOT THAT CYBERBOT REVERTED (CHRONOLOGICAL ORDER)
==


* rev id: 876258931 - (according to the edit summary) sigmabot inserted a heading template, and then cyberbot cleared the info on this sandbox (didn't clear the  template, so don't know why its labed as revert)

* rev id: 876259686 - same edit summary as before sigmabot added the sandbox header and, then cyberbot removed another user's revisions on that page, leaving the sandbox header that sigmabot included

*  rev id: 876282709 - same edit summary sigmabot doing what it does and then cyberbot, removes the info on that page made by another user (including a space that  sigma bot included - maybe this is why its a revert)

*  rev id: 876302133 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between) and the space that sigmabot included

*  rev id: 876468956 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between) and the space that sigmabot included

*  rev id: 876573806 - sigmabot doing the same thing and then cyberbot changes a couple small things of sigmabots template (removes)


*  rev id: 876652999 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between) and some syntax that sigmabot included


*  rev id: 876904050 -  sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and some syntax that sigmabot included


*  rev id: 876926815 - revision id linked to a page where a user was removing sigmabot edits and then, later cyberbot, removes an edit that that user included before? (or another user), leaving the template sigmabot included...

*  rev id: 876926998 - same edit summary as before, this revision happened after one above. A user removed sigmabots templates, sigmabot added it again, and then cyberbot removed that users revisions and a space that sigmatbot included

*  rev id: 876932206 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and some syntax that sigmabot included

*  rev id: 876936211 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before), leaving the heading sigmabot included

*  rev id: 877396169 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and a *space* that sigmabot included

*  rev id: 877400927 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and a *space* that sigmabot included

*  rev id: 877464155 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) 

*  rev id: 877505333 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and a *space* that sigmabot included

*  rev id: 877543107 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened before) and a *space* that sigmabot included

*  rev id: 877624662 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened before) and a *space* that sigmabot included

*  rev id: 877671439 - sigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and a *space* that sigmabot included

*  rev id: 877732494 - Tsigmabot doing the same thing and then cyberbot removes edits by other users (that happened in between/before) and a *space* that sigmabot included

NOTES: seems like mostly like collaborative work!


sigmabot was requested for approval on 4/29/2012 and was speedily approved, only has one task which is to reinstate the sandbox header (the main wikipedia sandbox)!

# are all these reverts in the sandbox namespace?

In [15]:
len(cyberbot_lsigmabot[cyberbot_lsigmabot.page_namespace == 4]) / len(cyberbot_lsigmabot)

0.9344262295081968

In [16]:
cyberbot_lsigmabot[cyberbot_lsigmabot.page_namespace != 4]

rev_id              user_text  revert_id  page_namespace  \
1069  876573806  Lowercase sigmabot II  876618768              10   
3512  878326891  Lowercase sigmabot II  878368216              10   
4412  878930050  Lowercase sigmabot II  878937026               2   
4637  879069597  Lowercase sigmabot II  879096318               5   

      rev_id_reverter user_text_reverter  
1069        876618768         Cyberbot I  
3512        878368216         Cyberbot I  
4412        878937026         Cyberbot I  
4637        879096318         Cyberbot I

https://en.wikipedia.org/wiki/Wikipedia:Namespace

876573806 in sandbox template, sigmabot adds template, cyberbot removes noinclude bit. apparently template sandbox allows you to experiment with editing templates.

878326891  same as last one

878930050 on User:Sandbox page sigmabot adds template, cyberbot clears content later

879069597 on wikipedia talk:sandbox, sigmabot adds template, cyberbot clears content later


perhaps the reason for these ones 